ember-client/src/main/scala/org/http4s/ember/client/internal/ClientHelpers.scala

```diff
@@ -72,13 +74,19 @@ private[client] object ClientHelpers {
             tlsContext
               .client(
                 initSocket,
-                TLSParameters(serverNames = Some(List(new SNIHostName(address.getHostName)))))
+                TLSParameters(serverNames = extractHostname(address.host).map(List(_))))
               .widen[Socket[F]]
           }
         else initSocket.pure[Resource[F, *]]
       }
     } yield RequestKeySocket(socket, requestKey)
 
+  private def extractHostname(from: Host): Option[SNIHostName] = from match {
+    case hostname: Hostname => new SNIHostName(hostname.normalized.toString).some
+    case address: IpAddress => new SNIHostName(address.toString).some
```

Comment:
Here it could be None, if we don't want SNI for IP based addresses.

Reply:
I am not an SNI expert, but a quick reading of RFC6066 suggests we don't?

> "HostName" contains the fully qualified DNS hostname of the server,
   as understood by the client.  The hostname is represented as a byte
   string using ASCII encoding without a trailing dot.  This allows the
   support of internationalized domain names through the use of A-labels
   defined in [RFC5890].  DNS hostnames are case-insensitive.  The
   algorithm to compare hostnames is described in [RFC5890], Section
   2.3.2.4.

> Literal IPv4 and IPv6 addresses are not permitted in "HostName".